### DISCUSSION TITLE
The notification do not expire even it has apns-expiration set to 0 in headers

### DISCUSSION THREAD

**Discussion by gedsakal** [relayrides/pushy#954] [Q&A] [answered]
Hi there,

I'm facing an issue while sending SimpleApnsPushNotification.

`
SimpleApnsPushNotification pushNotification = new SimpleApnsPushNotification(token,  topic,  payload,  null);
`

While debugging the code I see the value in DefaultHttp2Headers set correctly:
`
DefaultHttp2Headers[
:method: POST, 
:authority: api.push.apple.com, 
:path: /3/device/fd3b7.....9, 
:scheme: https, 
apns-expiration: 0, 
apns-priority: 10, 
apns-topic: com.b...x.devel.voip, 
authorization: bearer ey.....
]
`

Before sending it I switch off the internet connection of my phone, execute the java code, and switch-on internet connection back after some time.... and then the Notification arrives. 

As explained in documentation I expect the message to expire after the first attempt to reach the mobile, but it is not.

Has it something to do with the topic ending: .voip?

Any suggestion, please?

Thanks

**Comment by jchambers** [comment, marked as answer]
The store-and-forward behavior (the part that decides what to do with the expiration time) is controlled by Apple's servers, and is outside of Pushy's control. That said, [Apple's documentation](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/sending_notification_requests_to_apns) does make it sound like this behavior is expected:

> …If the [`apns-expiraton`] value is 0, APNs attempts to deliver the notification only once and doesn’t store it.
>
> A single APNs attempt may involve retries over multiple network interfaces and connections of the destination device. Often these retries span over some time period, depending on the network characteristics. In addition, a push notification may take some time on the network after APNs sends it to the device. APNs uses best efforts to honor the expiry date without any guarantee. If the value is nonzero, the notification may be delivered after the mentioned date. If the value is 0, the notification may be delivered with some delay.

For more specific questions, I'm afraid you'd have to ask somebody at Apple.
